pkg/types/azure/platform.go

```diff
@@ -35,6 +35,8 @@ type Platform struct {
 	ClusterOSImage string `json:"clusterOSImage,omitempty"`
 
 	// BaseDomainResourceGroupName specifies the resource group where the Azure DNS zone for the base domain is found.
+	//
+	// +optional
 	BaseDomainResourceGroupName string `json:"baseDomainResourceGroupName,omitempty"`
```

Comment:
You could probably write an explanation why this is parameter is optional and under what circumstances it should be set/not set.